### PR TITLE
setting image for container in generator

### PR DESCRIPTION
**Pull request by manute**
https://github.com/InVisionApp/engineering-velocity/issues/712

### CODE REVIEW

File: src/lib/generator.js [hunk 1]
```diff
@@ -301,6 +271,60 @@ class Generator {
...
+    if (artifact.image) {
+      this.eventHandler.emitWarn(
+        `Image ${artifact.image} already defined for ${artifact.name}`
+      );
+      return;
+    }
+
+    // Use the commitId when passed in
+    if (this.options.commitId) {
+      artifact.image = this.options.commitId;
```
[the remark this answers is not in the document]
Comment (manute) [reply src/lib/generator.js]: ah yes 🤦‍♂️

File: src/lib/generator.js [hunk 2]
```diff
@@ -301,6 +271,60 @@ class Generator {
     }).bind(this)();
   }
 
+  /**
+   * Sets the image tag , taking priority the SHA passed in
+   * @param	{[type]} artifact	  	 The clone of the container
+   * @param	{[type]} localBranch	 Local branch
+   * @return will throw an error if it's not valid tag for branch
+   */
+  setImageFor(artifact, localBranch) {
```
[the remark this answers is not in the document]
Comment (manute) [reply src/lib/generator.js]: Well I just changed the logic inside the`for` as it had duplicated things and was very complex to follow.
 Also I have refactored using the tests existed and also created new ones.

[the remark this answers is not in the document]
Comment (c1freitas) [reply src/lib/generator.js]: RIght, it's the _changed the logic_ that worries me. We don't want to take a chance at introducing any bugs if we don't have to, especially since this is legacy code and will get retired, hopefully soon. I know for a fact those tests don't cover every edge case, but we do know this logic has worked for the last year, so I am not a fan of changing it unless it's really needed.

**Comment by manute** [comment]
ok @c1freitas  modified the logic inside the `for` to address the image tag SHA only

File: src/lib/generator.js [hunk 3]
```diff
@@ -254,30 +254,47 @@ class Generator {
...
-            if (
-              !this.options.imageResourceDefs[artifact.image_tag] ||
-              !this.options.imageResourceDefs[artifact.image_tag][
-                artifactBranch
-              ]
-            ) {
+          if (this.options.commitId) {
+            localConfig[
+              containerName
+            ].image = `quay.io/invision/${artifact.name}:release-${this.options
```
Comment (c1freitas) [inline src/lib/generator.js]: This needs to use the `quay.io/${artifact.image_tag}` not `quay.io/invision/${artifact.name}`And if not found it should probably fall through to the final `else`. This can be different than the resource name and the github repo name, for example: https://github.com/InVisionApp/kubernetes-coreos/blob/master/manifests/base-cluster.yaml#L574

Comment (manute) [reply src/lib/generator.js]: ah ok

Comment (manute) [reply src/lib/generator.js]: done
